internal/report/userfriendlyprinter.go

```diff
@@ -0,0 +1,196 @@
+// Copyright 2023 Adevinta
+
+package report
+
+import (
+	"fmt"
+	"io"
+	"maps"
+	"strings"
+	"text/template"
+
+	"github.com/fatih/color"
+
+	"github.com/adevinta/lava/internal/config"
+)
+
+// userPrinter represents a user-friendly report printer.
+type userPrinter struct{}
+
+// returnColorFunc returns a function that colors a text.
+func returnColorFunc(colorAttribute color.Attribute) func(format string, a ...interface{}) string {
+	return func(format string, a ...interface{}) string {
+		attr := color.New(colorAttribute)
+		if len(a) == 0 {
+			return attr.Sprint(format)
+		}
+		return attr.Sprintf(format, a...)
+	}
+}
+
+// colorFuncs stores common function to colorize texts.
+var colorFuncs = template.FuncMap{
+	"magenta": returnColorFunc(color.FgMagenta),
+	"red":     returnColorFunc(color.FgRed),
+	"yellow":  returnColorFunc(color.FgYellow),
+	"cyan":    returnColorFunc(color.FgCyan),
+	"white":   returnColorFunc(color.FgWhite),
+	"bold":    returnColorFunc(color.Bold),
+}
+
+// Print renders the scan results in user-friendly format.
+func (prn userPrinter) Print(w io.Writer, vulns []vulnerability, sum summary) error {
+	if err := printSummary(w, sum); err != nil {
+		return fmt.Errorf("print summary: %w", err)
+	}
+	if len(vulns) > 0 {
+		_, err := fmt.Fprint(w, "\nVulnerabilities details:\n")
+		if err != nil {
+			return fmt.Errorf("print vulnerability details: %w", err)
+		}
+	}
+	for _, v := range vulns {
+		if err := printVulnerability(w, v); err != nil {
+			return fmt.Errorf("print vulnerability: %w", err)
+		}
+	}
+	return nil
+}
+
+// printSummary renders the summary table with the vulnerabilities stats.
+func printSummary(writer io.Writer, sum summary) error {
+	var total int
+	// count the total non-excluded vulnerabilities found.
+	for _, ss := range sum.count {
+		total = +ss
+	}
+
+	type severityCount struct {
+		Name  string
+		Count int
+	}
+
+	var sevCounts []severityCount
+	for sev := config.SeverityCritical; sev >= config.SeverityInfo; sev-- {
+		sevCount := severityCount{
+			Name:  sev.String(),
+			Count: sum.count[sev],
+		}
+		sevCounts = append(sevCounts, sevCount)
+	}
+
+	data := struct {
+		SevCounts []severityCount
+		Total     int
+		Excluded  int
+		Colors    map[string]string
+	}{
+		SevCounts: sevCounts,
+		Total:     total,
+		Excluded:  sum.excluded,
+	}
+
+	funcMap := template.FuncMap{
+		"upper": strings.ToUpper,
+	}
+	maps.Copy(funcMap, colorFuncs)
+	sumT := template.New("summary")
+	sumT, err := sumT.Funcs(funcMap).Parse(`
```

Comment:
What about this simplification?

```suggestion
	sumTmpl := template.New("summary").Funcs(funcMap).Funcs(colorFuncs)
	sumTmpl = template.Must(sumTmpl.Parse(`...`))
```

I would just panic if the template cannot be parsed. At the end of the day, there isn't anything you can do with the error.

Also, it is common to use the abbreviation `tmpl` for templates (e.g. `sumTmpl`).